geopandas/_vectorized.py

```diff
@@ -0,0 +1,853 @@
+"""
+Compatibility shim for the vectorized geometry operations.
+
+Uses PyGEOS if available/set, otherwise loops through Shapely geometries.
+
+"""
+import warnings
+
+import numpy as np
+
+import shapely.geometry
+import shapely.wkb
+import shapely.wkt
+
+from shapely.geometry.base import BaseGeometry
+
+from . import _compat as compat
+
+try:
+    import pygeos
+except ImportError:
+    geos = None
+
+
+_names = {
+    "NAG": None,
+    "POINT": "Point",
+    "LINESTRING": "LineString",
+    "LINEARRING": "LinearRing",
+    "POLYGON": "Polygon",
+    "MULTIPOINT": "MultiPoint",
+    "MULTILINESTRING": "MultiLineString",
+    "MULTIPOLYGON": "MultiPolygon",
+    "GEOMETRYCOLLECTION": "GeometryCollection",
+}
+
+if compat.USE_PYGEOS:
+    type_mapping = {p.value: _names[p.name] for p in pygeos.GeometryType}
+else:
+    type_mapping = None
+
+
+def _isna(value):
+    """
+    Check if scalar value is NA-like (None or np.nan).
+
+    Custom version that only works for scalars (returning True or False),
+    as `pd.isna` also works for array-like input returning a boolean array.
+    """
+    if value is None:
+        return True
+    elif isinstance(value, float) and np.isnan(value):
+        return True
+    else:
+        return False
+
+
+def _pygeos_to_shapely(geom):
+    if geom is None:
+        return None
+    elif pygeos.is_empty(geom) and pygeos.get_type_id(geom) == 0:
+        return shapely.wkt.loads("POINT EMPTY")
```

Comment:
I would add a comment explaining why empty point is separately, similar (but shorter) to one in tests.